src/main/java/software/amazon/awssdk/crt/s3/S3Client.java

```diff
@@ -64,11 +64,11 @@ public S3MetaRequest makeMetaRequest(S3MetaRequestOptions options) {
         if (options.getCredentialsProvider() != null) {
             credentialsProviderNativeHandle = options.getCredentialsProvider().getNativeHandle();
         }
-
         URI endpoint = options.getEndpoint();
+        int checksumAlgorithm = options.getChecksumAlgorithm() != null ? options.getChecksumAlgorithm().getNativeValue() : 0;
```

Comment:
do we need this? Will C part just parse null as zero if you pass null into C? (I am not sure about the answer though, just curious)

Reply:
As written this would be a type error as s3ClientMakeMetaRequest takes an int as an argument. This is how we handle enums to jni, we extract the int that the java level. It seems other people do this differently though, not sure what the tradeoffs are though. https://community.oracle.com/tech/developers/discussion/1549037/passing-enums-as-arguments-within-a-jni-function

Reply:
do we have an enum value for 0? this is a bit hard to read. 

Can also do something like Optional.ofNullable(getAlgo).map(Algo::getNativeValue).orElse(0)